src/components/hv-picker-field/index.js

```diff
@@ -0,0 +1,247 @@
+   */
+  cancelPickerModal = () => {
+    this.setState({
+      showPicker: false,
+    });
+  };
+
+  /**
+   * Hides the picker and applies the chosen value to the field.
+   */
+  savePickerModal = () => {
+    const { element } = this.props;
+    this.setState({
+      showPicker: false,
+      value: this.state.pickerValue,
+    });
+    element.setAttribute('value', this.state.pickerValue || '');
+  };
+
+  /**
+   * Renders the picker component. Picker items come from the
+   * <picker-item> elements in the <picker-field> element.
+   */
+  renderPicker = () => {
+    const { element } = this.props;
+    const props = {
+      onValueChange: value => {
+        this.setState({ pickerValue: value });
+      },
+      selectedValue: this.state.pickerValue,
+    };
+
+    // Gets all of the <picker-item> elements. All picker item elements
+    // with a value and label are turned into options for the picker.
+    const children = Array.from(
+      element.getElementsByTagNameNS(
+        Namespaces.HYPERVIEW,
+        LOCAL_NAME.PICKER_ITEM,
+      ),
+    ).map(item => {
+      if (!item) {
+        return null;
+      }
+      const label = item.getAttribute('label');
+      const value = item.getAttribute('value');
+      if (!label || value === null) {
+        return null;
+      }
+      return React.createElement(Picker.Item, { label, value });
+    });
+
+    return React.createElement(Picker, props, ...children);
+  };
+
+  /**
+   * Renders a bottom sheet with cancel/save buttons and a picker component.
+   * Uses styles defined on the <picker-field> element for the modal and buttons.
+   */
+  renderPickerModal = () => {
+    const { element, stylesheets, options } = this.props;
+    const modalStyle = createStyleProp(element, stylesheets, {
+      ...options,
+      styleAttr: 'modal-style',
+    });
+    const cancelTextStyle = createStyleProp(element, stylesheets, {
+      ...options,
+      pressed: this.state.cancelPressed,
+      styleAttr: 'modal-text-style',
+    });
+    const saveTextStyle = createStyleProp(element, stylesheets, {
+      ...options,
+      pressed: this.state.savePressed,
+      styleAttr: 'modal-text-style',
+    });
+    const cancelLabel = element.getAttribute('cancel-label') || 'Cancel';
+    const saveLabel = element.getAttribute('save-label') || 'Save';
+
+    return (
+      <Modal
+        animationType="slide"
+        transparent
+        visible={this.state.showPicker}
+        onRequestClose={this.cancelPickerModal}
+      >
+        <View
+          style={{
+            flex: 1,
+            justifyContent: 'flex-end',
+          }}
+        >
+          <View style={modalStyle}>
+            <View
+              style={{
+                flexDirection: 'row',
+                justifyContent: 'space-between',
+              }}
+            >
+              <TouchableWithoutFeedback
+                onPressIn={() => this.setState({ cancelPressed: true })}
```

Comment:
You can create a toggle state function for `cancelPressed` that would inverse the state